src/app/g3w-ol/projection/projections.js

```diff
@@ -1,104 +1,55 @@
-const Projection = require('g3w-ol/projection/projection');
-const ADDEDPROJECTIONS = ['EPSG:4326', 'EPSG:3857'];
+import {API_BASE_URLS} from 'constant';
+const Projection = require('./projection');
+const {XHR} = require('core/utils/utils');
+const {normalizeEpsg} = require('core/utils/geo');
 
 const Projections = {
+
+  /**
+   * @since v3.8
+   */
+  isRegistered(epsg) {
+    return ol.proj.get(epsg);
+  },
+
   get(crs={}) {
-    const cachedProjection = ADDEDPROJECTIONS.indexOf(crs.epsg) !== -1 ?  ol.proj.get(crs.epsg) : null;
+    const cachedProjection = this.isRegistered(crs.epsg);
     if (cachedProjection) return cachedProjection;
     const projection = new Projection({
       crs
     });
     ol.proj.addProjection(projection);
-    ADDEDPROJECTIONS.push(crs.epsg);
+    ol.proj.proj4.register(proj4);
     return projection;
   },
+
   /**
-   * extent get from https://epsg.io/
+   * Check and register epsg
+   * 
+   * @param epsg : "EPSG:<CODE>" Ex. "EPSG:4326"
+   * @returns {Promise<unknown>}
```

Comment:
Double check too, but I think the fulfilled type should be as follows:

```suggestion
   * @returns {Promise<Projection>}
```

Ref:

https://github.com/g3w-suite/g3w-client/blob/2affa0475d081d7f366dd5df3c33e8b18aac0d10/src/app/g3w-ol/projection/projections.js#L45